python/housinginsights/ingestion/Manifest.py

```diff
@@ -0,0 +1,135 @@
+                    row['data_date'] = datetime.strftime(_date, '%Y-%m-%d')
+                except ValueError:
+                    row['data_date'] = 'Null'
+
+                #return the row
+                yield row
+
+    def has_unique_ids(self):
+        """
+        Verifies that every value in the manifest column 'unique_data_id' is
+        in fact unique. This makes sure that we can rely on this column to
+        uniquely identify a source CSV file, and and can connect a record in
+        the SQL manifest to the manifest.csv
+
+        :return: true if passes validation; false otherwise
+        """
+        self.unique_ids = {}
+
+        for row in self:
+            if row['unique_data_id'] in self.unique_ids:
+                return False
+            else:
+                #don't add flags that won't make it into the SQL database
+                if row['include_flag'] in Manifest._include_flags_positive:
+                    self.unique_ids[row['unique_data_id']] = 'found'
+        return True
+
+    def get_manifest_row(self, unique_data_id):
+        """
+        Returns the row for the given unique_data_id else 'None' if not in
+        manifest.
+        """
+        for row in self:
+            use = row['include_flag'] == 'use'
+            if row['unique_data_id'] == unique_data_id and use:
+                return row
+        return None
+
+    def create_list(self, folder_path):
+        """
+        Returns a list of potential unique_data_ids based on csv files in the
+        given folder path. The assumption is that all raw data a saved as csv
+        with filename that matches unique_data_id used in manifest.
+        """
+        unique_data_ids = list()
+        files_in_folder = os.listdir(folder_path)
+        for file in files_in_folder:
+            file_path = os.path.join(folder_path, file)
+            uid, file_ext = file.split(sep='.')
+            if os.path.isfile(file_path) and file_ext == 'csv':
+                unique_data_ids.append(uid)
+
+        return unique_data_ids
+
+    def update_manifest(self, date_stamped_folder):
+        timestamp = os.path.basename(date_stamped_folder)
+        data_date = datetime.strptime(timestamp, '%Y%m%d').strftime('%m/%d/%Y')
+        file_path_base = date_stamped_folder[date_stamped_folder.find('raw'):]
+
+        field_names = self.keys
+
+        # get unique_data_ids for data files in recent subfolder
+        uid_list = self.create_list(date_stamped_folder)
+        data = list()
+
+        # update the manifest.csv in place
+        for row in self:
+            row_uid = row['unique_data_id']
+            if row_uid in uid_list:
+                row['data_date'] = data_date
+                filepath = os.path.join(file_path_base,
+                                        "{}.csv".format(row_uid))
+                row['filepath'] = filepath
+            else:
+                # parse the date back into format used in manifest
```

Comment:
@jkwening can you elaborate? If we change the hardcoded values to YYYY-MM-DD can we remove this section and just do data.append(row) directly?

I think it actually would work this way regardless if we just remove this else block - as soon as we run it the first time, the manifest.csv would then have the year-first date formats, but the `__iter__` method dateparser I think is generic enough to handle either way?

The reason the date format is mm/dd/yyyy is because Excel automatically puts them that way, and I'm usually using Excel to manually edit the manifest. So `__iter__` needs to be able to handle either way in case the file is opened and accidentally saved with month first format, but when auto-writing year first format can certainly be default